gulpfile.js

```diff
@@ -0,0 +1,239 @@
+const argv = require('yargs');
+const bump = require('gulp-bump');
+const exec = require('child_process').exec;
+const del = require('del');
+const es = require('event-stream');
+const fs = require('file-system');
+const git = require('gulp-git');
+const gulp = require('gulp');
+const debug = require('gulp-debug');
+const inlinesource = require('gulp-inline-source');
+const moment = require('moment');
+const shell = require('gulp-shell');
+
+//Parameters and Variables
+const params = require('./build/parameters');
+let lerna = require('./lerna.json');
+
+function adoPrep(done) {
+    var branchName = params.build.buildSourceBranch;
+    if (branchName !== 'master') {        
+        branchName = branchName.replace(/refs\/heads\/(feature\/)?/i, '');
+    }    
+    var gitScript = `sudo git checkout ${branchName}`;
+    console.log('ADO Prep Script: ' + gitScript);
+    return runCmd(gitScript, done);    
+}
+
+function bumpVersion() {
+    return gulp.src(['ingredient/**/package.json','core/package.json','system/package.json','lerna.json'], { base: './' })
+        .pipe(bump({ type: 'patch' }))
+        .pipe(gulp.dest('./'));
+}
+
+function build(done) {
+    switch(true) {
+        case ( !params.conditions.isRunningOnADO ): 
+            console.log('Running Local Build');
+            gulp.series(lernaBuild)(done);
+        break;
+
+        case ( params.conditions.isRunningOnADO && 
+                !params.build.pullRequestID && 
+                !!params.build.buildSourceBranch.replace(/refs\/heads\/(feature\/)?/i, '').match(/master/ig)):
+            console.log('Running Azure DevOps Release Build');
+            gulp.series( printVersion, adoPrep, toolInstall, lernaBuild, lernaPublish, systemPublish, gitCommit )(done);
+        break;
+
+        case ( params.conditions.isRunningOnADO && 
+                !!params.build.pullRequestID ):
+            console.log('Running Azure DevOps Pull Request Build');
+            gulp.series( printVersion, adoPrep, toolInstall, lernaBuild )(done);
+        break;
+        
+        case ( params.conditions.isRunningOnADO && 
+            params.build.buildReason.match(/manual/ig) ):
+            console.log('Running Azure DevOps Manual Build');
+            gulp.series( printVersion, adoPrep, toolInstall, lernaBuild )(done);
+        break;
+
+        default:
+            done('Build did not complete!');
+        break;        
+    }
+}
+
+function cleanCoverage() {
+    return del('coverage/**', { force: true });
+}
+
+
+function gitCommit(done) {
```

Comment:
lerna already creates the chore update/publish commit, and does the push. Not sure we need this?